packages/flutter_tools/test/general.shard/args_test.dart

```diff
@@ -22,6 +24,24 @@ void main() {
     ).forEach(runner.addCommand);
     verifyCommandRunner(runner);
   }));
+
+  testUsingContext('bool? safe argResults', () async {
+    final DummyFlutterCommand command = DummyFlutterCommand(
+        commandFunction: () async {
+          return const FlutterCommandResult(ExitStatus.success);
+        }
+    );
+    final FlutterCommandRunner runner = FlutterCommandRunner(verboseHelp: true);
+    command.argParser.addFlag('key');
```

Comment:
Let's add 2 more expected

```
command.argParser.addFlag('key-false');
await runner.run(<String>['dummy', '--key']);
expect(command.boolArg('key-false'), false);
expect(command.boolArgDeprecated('key-false'), false);
```

I believe that with your current code key-false will map to null in boolArg without my previous comment